core-domain/src/main/java/org/uniprot/core/uniprotkb/impl/UniProtKBEntryBuilder.java

```diff
@@ -174,21 +180,39 @@ private UniProtKBEntryBuilder(
 
     public @Nonnull UniProtKBEntryBuilder commentsAdd(Comment comment) {
         addOrIgnoreNull(comment, this.comments);
+        putIfNameAndValueNotNull(
```

Comment:
Maybe, instead of add this logic in many places, we could add in one single place (the build method for all comments and features). This way would make the code easier to maintain. (reducing a single point of failure and single point test as well). And also it would improve the performance avoiding iterate over all collection every time we add a new comment. (createCountByCommentTypeMap does the iteration)